### PR TITLE
Add cheat recipe

### PR DESCRIPTION
**Pull request by proinsias**

### CODE REVIEW

File: recipes/cheat/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+test:
+  imports:
+    - cheat
+    - cheat.cheatsheets
+    - cheat.test
+
+about:
+  home: https://github.com/chrisallenlane/cheat
+  license: GPL3
+  # license_file: LICENSE # When added to MANIFEST.in – https://github.com/chrisallenlane/cheat/pull/291
```
Comment (jakirkham) [inline recipes/cheat/meta.yaml]: xref: https://github.com/chrisallenlane/cheat/pull/291

File: recipes/cheat/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{% set name = "cheat" %}
+{% set version = "2.1.9" %}
+{% set md5 = "3bf4c63c194c493040c5a1b0f8e5c0dc" %}
```
Comment (jakirkham) [inline recipes/cheat/meta.yaml]: Could switch to `sha256` using [pypi.io]( https://pypi.io/ ).

File: recipes/cheat/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  md5: {{ md5 }}
+
+build:
+  number: 0
+  script: python setup.py install
+  skip: True  # [win and py3k]
+
+requirements:
+  build:
+    - python
```
Comment (jakirkham) [inline recipes/cheat/meta.yaml]: Can we drop the dependencies below from the build requirements or are they needed during the build for some reason?

File: recipes/cheat/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    - cheat
+    - cheat.cheatsheets
+    - cheat.test
+
+about:
+  home: https://github.com/chrisallenlane/cheat
+  license: GPL3
+  # license_file: LICENSE # When added to MANIFEST.in – https://github.com/chrisallenlane/cheat/pull/291
+  summary: 'cheat allows you to create and view interactive cheatsheets on the command-line. It was designed to help remind *nix system administrators of options for commands that they use frequently, but not frequently enough to remember.'
+  license_family: GPL3
```
Comment (jakirkham) [inline recipes/cheat/meta.yaml]: GPL? @goanpeca, how do we do this?

**Comment by goanpeca** [comment]
what part of gpl worries you?

**Comment by jakirkham** [comment]
No worries. That was a little too terse of a question on my part. Sorry about that.

File: recipes/cheat/meta.yaml (hunk 4 above)
Comment (goanpeca) [inline recipes/cheat/meta.yaml]: I would say yes GPL

**Comment by jakirkham** [comment]
Weird. Apparently GPL is not an allowed family. One must specify the version of GPL too. 😕 I don't think I understand the point of this field with this constraint.

```
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build-all", line 11, in <module>
    load_entry_point('conda-build-all==1.0.0rc1', 'console_scripts', 'conda-build-all')()
  File "/opt/conda/lib/python3.5/site-packages/conda_build_all/cli.py", line 90, in main
    b.main()
  File "/opt/conda/lib/python3.5/site-packages/conda_build_all/builder.py", line 248, in main
    recipe_metas = self.fetch_all_metas(build_config)
  File "/opt/conda/lib/python3.5/site-packages/conda_build_all/builder.py", line 163, in fetch_all_metas
    recipe_metas = list_metas(conda_recipes_directory, config=config)
  File "/opt/conda/lib/python3.5/site-packages/conda_build_all/builder.py", line 85, in list_metas
    packages.append(MetaData(new_root))
  File "/opt/conda/lib/python3.5/site-packages/conda_build/metadata.py", line 373, in __init__
    self.parse_again(permit_undefined_jinja=True)
  File "/opt/conda/lib/python3.5/site-packages/conda_build/metadata.py", line 387, in parse_again
    self.meta = parse(self._get_contents(permit_undefined_jinja), path=self.meta_path)
  File "/opt/conda/lib/python3.5/site-packages/conda_build/metadata.py", line 188, in parse
    ensure_valid_license_family(res)
  File "/opt/conda/lib/python3.5/site-packages/conda_build/metadata.py", line 159, in ensure_valid_license_family
    (license_family, comma_join(sorted(allowed_license_families)))))
RuntimeError: about/license_family 'GPL' not allowed. Allowed families are AGPL,
Apache, BSD, GPL2, GPL3, LGPL, MIT, Other, PSF, Proprietary, and
Public-Domain.
```

File: recipes/cheat/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  imports:
+    - cheat
+    - cheat.cheatsheets
+    - cheat.test
+
+about:
+  home: https://github.com/chrisallenlane/cheat
+  license: GPL3
+  # license_file: LICENSE # When added to MANIFEST.in – chrisallenlane/cheat#291
+  license_family: GPL
```
Comment (jakirkham) [inline recipes/cheat/meta.yaml]: So I guess this must be `GPL3`. 😖 Alternatively I'm ok with it being dropped as it is redundant.

**Comment by jakirkham** [comment]
Raised issue ( https://github.com/conda/conda-build/issues/1300 ) about the `license_family` constraint that requires license version for GPL licenses.

File: recipes/cheat/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/cheat/meta.yaml]: Done.

File: recipes/cheat/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install
+  skip: True  # [win and py3k]
```
Comment (jakirkham) [inline recipes/cheat/meta.yaml]: Sorry have forgotten. Why the skip?

Comment (proinsias) [inline recipes/cheat/meta.yaml]: Due to a Windows/Python 3 [issue](aerkalov/ebooklib#104) with the ebooklib dependency, skip the Windows/Python 3 builds for now.

File: recipes/cheat/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+test:
+  imports:
+    - cheat
+    - cheat.cheatsheets
+    - cheat.test
+
+about:
+  home: https://github.com/chrisallenlane/cheat
+  license: GPL3
+  # license_file: LICENSE # When added to MANIFEST.in – chrisallenlane/cheat#291
```
Comment (jakirkham) [inline recipes/cheat/meta.yaml]: xref: https://github.com/chrisallenlane/cheat/pull/291

**Comment by proinsias** [comment]
@jakirkham – need anything else from me?

File: recipes/cheat/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+test:
+  imports:
+    - cheat
+    - cheat.cheatsheets
+    - cheat.test
+
+about:
+  home: https://github.com/chrisallenlane/cheat
+  license: GPL3
```
Comment (jakirkham) [inline recipes/cheat/meta.yaml]: Apparently dual licensed. So `MIT/GPL3`.

File: recipes/cheat/meta.yaml (hunk 6 above)
Comment (jakirkham) [inline recipes/cheat/meta.yaml]: I'm confused. Where does this require `ebooklib`?

File: recipes/cheat/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+test:
+  imports:
+    - cheat
+    - cheat.cheatsheets
+    - cheat.test
+
+about:
+  home: https://github.com/chrisallenlane/cheat
+  license: MIT/GPL3
+  # license_file: LICENSE # When added to MANIFEST.in – chrisallenlane/cheat#291
```
Comment (jakirkham) [inline recipes/cheat/meta.yaml]: xref: https://github.com/chrisallenlane/cheat/pull/291

**Comment by jakirkham** [comment]
Nope. Thanks @proinsias. 😄

**Pull request merged**